core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -4818,10 +4818,10 @@ object ZIO extends ZIOCompanionPlatformSpecific with ZIOCompanionVersionSpecific
    * not use this method, use [[ZIO.suspend]].
    */
   def suspendSucceed[R, E, A](zio: => ZIO[R, E, A])(implicit trace: Trace): ZIO[R, E, A] =
-    ZIO.succeed(zio).flatMap(identityFn)
+    ZIO.FlatMap(trace, ZIO.unit, (_: Any) => zio)
```

Comment:
```suggestion
    ZIO.unit.flatMap((_: Any) => zio)
```

Reply:
Why do you think it's better? 🤔
Feels like adding an additional step, which is what we try to avoid as much as possible

Reply:
Seems more readable to me. Given the size of the function body, I don't see why JIT wouldn't inline.

Reply:
To me, at this level (lowest level of ZIO code), we shouldn't rely on the JIT when we can and when it's that simple.
Readability here is quite subjective. To me, the "direct approach" is more readable as I don't have to understand how `.flatMap` works to understand the code

Reply:
> To me, at this level (lowest level of ZIO code), we shouldn't rely on the JIT when we can and when it's that simple.

If that's the case, why are you using `FlatMap.apply`? Just instantiate the class with `new` 🙂 Single layer method indirection where the body is 1 line shouldn't be an issue...

There's a decent amount of precedence for using `unit.flatMap` FWIW: https://github.com/scala/scala/blob/2.13.x/src/library/scala/concurrent/Future.scala#L706-L707